packages/contracts/contracts/SmartEscrowMono.sol

```diff
@@ -113,124 +114,121 @@ contract SmartEscrowMono is Context, IArbitrable, ReentrancyGuard {
+    escrow.balance = escrow.balance.sub(amount);
 
     emit Release(index, currentMilestone, amount);
   }
 
   function withdraw(uint256 index) external nonReentrant {
-    // withdraw locker remainder to client if termination time passes & no lock
-    Locker storage locker = lockers[index];
+    // withdraw escrow balance to client if termination time passes & no lock
+    Escrow storage escrow = escrows[index];
 
-    require(locker.confirmed, "!confirmed");
-    require(!locker.locked, "locked");
-    require(locker.total > locker.released, "released");
-    require(block.timestamp > locker.terminationTime, "!terminated");
+    // require(escrow.confirmed, "!confirmed");
+    require(!escrow.locked, "locked");
+    require(escrow.balance > 0, "balance is zero");
+    require(block.timestamp > escrow.terminationTime, "!terminated");
 
-    uint256 remainder = locker.total.sub(locker.released);
-    IERC20(locker.token).safeTransfer(locker.client, remainder);
+    IERC20(escrow.token).safeTransfer(escrow.client, escrow.balance);
 
-    emit Withdraw(index, remainder);
+    emit Withdraw(index, escrow.balance);
   }
 
-  /************
-    ADR FUNCTIONS
-    ************/
   function lock(uint256 index, bytes32 details) external nonReentrant {
-    // client or main (0) provider can lock remainder for resolution during locker period / update request details
-    Locker storage locker = lockers[index];
+    // client or main (0) provider can lock balance for resolution during escrow period / update request details
+    Escrow storage escrow = escrows[index];
 
-    require(locker.confirmed, "!confirmed");
-    require(!locker.locked, "locked");
-    require(locker.total > locker.released, "released");
-    require(block.timestamp < locker.terminationTime, "terminated");
+    // require(escrow.confirmed, "!confirmed");
+    require(!escrow.locked, "locked");
+    require(escrow.balance > 0, "balance is zero");
+    require(block.timestamp < escrow.terminationTime, "terminated");
     require(
-      _msgSender() == locker.client || _msgSender() == locker.provider,
+      _msgSender() == escrow.client || _msgSender() == escrow.provider,
       "!party"
     );
 
-    if (locker.resolverType == ADR.ARAGON_COURT) {
-      IArbitrator arbitrator = IArbitrator(locker.resolver);
-      payDisputeFees(locker.resolver, locker);
+    if (escrow.resolverType == ADR.ARAGON_COURT) {
+      IArbitrator arbitrator = IArbitrator(escrow.resolver);
+      payDisputeFees(escrow.resolver, escrow);
       uint256 disputeId = arbitrator.createDispute(
         DISPUTES_POSSIBLE_OUTCOMES,
-        "0x"
+        abi.encodePacked(details)
       );
       disputes[disputeId] = index;
     }
-    locker.locked = true;
+    escrow.locked = true;
 
     emit Lock(_msgSender(), index, details);
   }
 
-  function payDisputeFees(address _adr, Locker storage locker) internal {
+  function payDisputeFees(address _adr, Escrow storage escrow) internal {
```

Comment:
since this is an internal function, should we name it `_payDisputeFees`? That's just a stylistic / convention thing; I don't have a strong preference either way ;)